### PR TITLE
fix(vuln): report architecture for `apk` packages

### PR DESCRIPTION
**Pull request by AliDatadog**
## Description

The architecture field was always an empty string for `apk` packages. This PR sets the field correctly.

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/handler/dpkg/dpkg_test.go
Comment (knqyf263) [inline pkg/fanal/handler/dpkg/dpkg_test.go:1]: What is this file for?

**Review by knqyf263 — COMMENTED** [review]
Thanks for your contribution. I left a question.

**Comment by knqyf263** [comment]
Yes, you can use `-update`.

```
go test -tags=integration ./pkg/fanal/test/integration/... -update 
```

File: pkg/fanal/test/integration/testdata/goldens/packages/alpine-310.json.golden [hunk 1]
```diff
@@ -3,131 +3,132 @@
...
     "Licenses": [
       "GPL-2.0"
     ],
-    "Digest": "sha1:574d490311b68db01c0a3e44f5491be0cdc79250",
     "DependsOn": [
       "busybox@1.30.1-r2",
       "musl@1.1.22-r3"
     ],
     "Layer": {
-      "Digest": "sha256:9d48c3bd43c520dc2784e868a780e976b207cbf493eaff8c6596eb871cbd9609",
```
Comment (knqyf263) [inline pkg/fanal/test/integration/testdata/goldens/packages/alpine-310.json.golden:17]: It is weird. It is gone.

**Review by knqyf263 — COMMENTED** [review]
@DmitriyLewen Can you take a look?

File: pkg/fanal/test/integration/testdata/goldens/packages/alpine-310.json.golden (hunk 1 above)
Comment (DmitriyLewen) [reply pkg/fanal/test/integration/testdata/goldens/packages/alpine-310.json.golden:17]: Looks like it is old issue, but all contributors manually updated golden files.
I continue to investigate issue.

Comment (DmitriyLewen) [reply pkg/fanal/test/integration/testdata/goldens/packages/alpine-310.json.golden:17]: I created #4379 to fix this problem.

Comment (knqyf263) [reply pkg/fanal/test/integration/testdata/goldens/packages/alpine-310.json.golden:17]: #4379 got merged.
@AliDatadog Could you merge the main branch and recreate the golden files again?

Comment (AliDatadog) [reply pkg/fanal/test/integration/testdata/goldens/packages/alpine-310.json.golden]: Thanks it seems to work now 👍

**Comment by AliDatadog** [comment]
It is unclear why the integration tests failed here.

**Comment by DmitriyLewen** [comment]
Hello @AliDatadog 
Looks like problem with `test-repo.json.golden` file.
This file is empty.

How did you update golden files? 
`mage test:updateGolden` command works correctly with this file on my local PC.

**Comment by AliDatadog** [comment]
Hi @DmitriyLewen, I tried to copy the `test-repo.json.golden` file from main and run `mage test:updateGolden`, it failed and deleted the file. I think the error is expected:
```
2023/05/19 12:38:13 Container is ready id: 33fef91ff319 image: registry:2
--- FAIL: TestTLSRegistry (2.64s)
    --- FAIL: TestTLSRegistry/happy_path (0.53s)
        registry_test.go:187: 
                Error Trace:    /Users/user/trivy/pkg/fanal/test/integration/registry_test.go:187
                Error:          Not equal: 
                                expected: false
                                actual  : true
                Test:           TestTLSRegistry/happy_path
                Messages:       2 errors occurred:
                                        * GET https://localhost:55689/v2/ghcr.io/aquasecurity/trivy-test-images/manifests/alpine-310: MANIFEST_UNKNOWN: manifest unknown; map[Tag:alpine-310]
                                        * GET https://localhost:55689/v2/ghcr.io/aquasecurity/trivy-test-images/manifests/alpine-310: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:ghcr.io/aquasecurity/trivy-test-images Type:repository]]
                            
    --- FAIL: TestTLSRegistry/happy_path_with_docker_login (0.27s)
        registry_test.go:179: 
                Error Trace:    /Users/user/trivy/pkg/fanal/test/integration/registry_test.go:179
                Error:          Received unexpected error:
                                exit status 1
                Test:           TestTLSRegistry/happy_path_with_docker_login
FAIL
FAIL    github.com/aquasecurity/trivy/pkg/fanal/test/integration        26.593s
FAIL
Error: running "go test -tags=integration ./integration/... ./pkg/fanal/test/integration/... -update" failed with exit code 1
```

I am running the tests with a Mac M1. Perhaps I need to install some more dependencies.

**Comment by DmitriyLewen** [comment]
hello @AliDatadog 
Do you have an error for this test only?

`test-repo.json.golden` file is only used [here](https://github.com/aquasecurity/trivy/blob/29b5f7e8ec2d89f70576118c3056bcc7c6246549/integration/client_server_test.go#L258-L265).

**Review by DmitriyLewen — APPROVED** [review]
LGTM

**Comment by knqyf263** [comment]
@AliDatadog Are you still facing any issues?

**Comment by AliDatadog** [comment]
Indeed, this test is the only one for which I have an error and `test-repo.json.golden` gets deleted by `mage: test:updateGolden`.
I think this PR is ready to be merged now @knqyf263

**Review by knqyf263 — APPROVED** [review]
Thanks for your efforts.

**Pull request merged**